src/java/org/apache/cassandra/service/accord/async/AsyncWriter.java

```diff
@@ -211,7 +212,8 @@ private void denormalize(AccordCommand command, AsyncContext context)
         }
 
         Map<PartitionKey, AccordCommandsForKey> addCfkToCtx = new HashMap<>();
-        if (CommandSummaries.commandsPerKey.needsUpdate(command))
+        // NotWitnessed has 0 state, and doesn't know about the tx keys, so unable to populate summary
+        if (command.status() != Status.NotWitnessed && CommandSummaries.commandsPerKey.needsUpdate(command))
```

Comment:
not actually sure this is the correct thing to do... a command may be dependent on a `NotWitnessed`, but not sure if this change is the proper change...